semgrep-core/src/analyzing/AST_to_IL.ml

```diff
@@ -711,6 +719,25 @@ let parameters _env params =
 (*****************************************************************************)
 (* Statement *)
 (*****************************************************************************)
+
+let mk_st_env_and_labels env tok =
+  let cont_label = fresh_label env tok in
+  let break_label = fresh_label env tok in
+  let st_env =
+    {
+      env with
+      break_labels = break_label :: env.break_labels;
+      cont_label = Some cont_label;
+    }
+  in
+  (* We add a dummy fixme stmt after each label in case the label ends up
```

Comment:
This would be a non-trivial change. The CFG builder currently associates labels to nodes, and doesn't include labels as their own nodes, which is why this happens. @IagoAbal do you think this is worth doing?

Reply:
[CIL](https://github.com/cil-project/cil) for example annotates statements with labels and it uses [dummy statements](https://github.com/cil-project/cil/blob/936b04103eb573f320c6badf280e8bb17f6e7b26/src/frontc/cabs2cil.ml#L1001) for `goto`s too. Although e.g. for `continue` I think they only add these labels on-demand.

Another option is to change the `IL.node` type and add a `NLabel` node, then change the CFG builder to use these nodes instead.

Reply:
I am not sure about pros/cons here, but given that in the IL we have a `Label` statement perhaps we should interpret it as `label: noop`. Then we can try to optimize this in the CFG builder, but create a dummy node by default for each label (in the CFG builder).

I suggest we add a `FIXME` note and create an issue to fix this (in some way) ASAP, but use a separate PR.

Reply:
Hmm, I looked into this more carefully and it may be easier to fix this in `CFG_build.cfg_stmt_list`. Basically, if we are done folding, and we have a bunch of unassigned labels, we just need to create a dummy node, connect `previ` (it any) to this node, and assigned the labels to it.

For the dummy nod we could use `NTodo` or `NOther` if we extend `other_stmt` with e.g. `Noop`.

Reply:
Ok yeah this actually was pretty easy. I added `Noop` to `other_stmt`. All the tests still pass with no complaints of unresolved gotos.